packages/interactivity/src/router.js

```diff
@@ -75,6 +75,9 @@ const renderRegions = ( page ) => {
 		const id = region.getAttribute( attrName );
 		const fragment = getRegionRootFragment( region );
 		render( page.regions[ id ], fragment );
+		if ( page.title ) {
+			document.title = page.title;
+		}
 	} );
```

Comment:
It's enough to change the title once per page; we don't need to do it once per region. 😄 

```suggestion
	} );
	if ( page.title ) {
		document.title = page.title;
	}
```